variable.go

```diff
@@ -232,3 +233,37 @@ func (v *Variable) Get(out any) error {
 
 	return nil
 }
+
+func checkVariable[T any](v *Variable) error {
+	if v.ReadOnly() {
+		return ErrReadOnly
+	}
+
+	var t T
+	size := binary.Size(t)
```

Comment:
Would it be clearer to check that `unsafe.Sizeof() == binary.Size()`?

Reply:
I think that would make things more complex, not less. The reader would have to be aware of the intricacies and the differences between both, and we'd still need the comparison to `v.size`, so it wouldn't eliminate any code.